### PR TITLE
Add users, groups and security configuration

### PR DESCRIPTION
**Pull request by matthijsln**
- [x] check if management security is enabled, maybe we should disable the port by default and only enable it in Docker Compose

### CODE REVIEW

File: src/main/java/nl/b3p/tailormap/api/persistence/User.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import org.hibernate.annotations.Type;
+
+@Entity
+@Table(name = "users")
+public class User {
```
Comment (mprins) [inline src/main/java/nl/b3p/tailormap/api/persistence/User.java:22]: Should we add a Date field to make it possible so set an expiry date for a user account? eg.
```suggestion
public class User {


  @FutureOrPresent
  private java.time.ZonedDateTime validUntil;
```

https://b3partners.atlassian.net/browse/HTM-114

**Review by mprins — APPROVED** [review]
Some notes.
I think we should start using Hibernate Validator now since we're redesiging the persistence and retrofitting will be more work; see https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/#section-builtin-constraints

I've added some to user

File: src/main/resources/import.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+insert into groups(name, system_group, title) values ('admin', true, 'Administrators with full access');
```
[the remark this answers is not in the document]
Comment (matthijsln) [reply src/main/resources/import.sql:2]: Let's try to keep localization as much in the (admin) frontend as possible... Can add some resourcebundles later but not too kind on maintaining that poilerblate for now...

File: src/main/java/nl/b3p/tailormap/api/persistence/Group.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  public static final String ADMIN = "admin";
+  public static final String ADMIN_GEO_SERVICES = "admin-geo-services";
+  public static final String ADMIN_USERS = "admin-users";
+  public static final String ADMIN_APPLICATIONS = "admin-applications";
+
+  @Id private String name;
+
+  private boolean systemGroup;
+
+  private String title;
```
[the remark this answers is not in the document]
Comment (matthijsln) [reply src/main/java/nl/b3p/tailormap/api/persistence/Group.java:28]: I had 'title' and 'notes', where 'title' (`varchar(255)`) would be displayed in a crud overview and notes (`text` column) in details.
We use title for most entities, i.e. GeoService, FeatureSource, Layer, etc.

File: src/main/java/nl/b3p/tailormap/api/persistence/User.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  @Id private String username;
+
+  private String password;
+
+  private String email;
```
[the remark this answers is not in the document]
Comment (matthijsln) [reply src/main/java/nl/b3p/tailormap/api/persistence/User.java:27]: 👍🏻 +1

File: src/main/java/nl/b3p/tailormap/api/persistence/User.java (hunk 1 above)
Comment (matthijsln) [reply src/main/java/nl/b3p/tailormap/api/persistence/User.java:26]: 👍🏻

File: src/main/java/nl/b3p/tailormap/api/persistence/Group.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthijsln) [reply src/main/java/nl/b3p/tailormap/api/persistence/Group.java:28]: For Groups a description makes more sense indeed.

**Pull request merged**